sinks/signalfx/signalfx.go

```diff
@@ -0,0 +1,107 @@
+package signalfx
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/datapoint/dpsink"
+	"github.com/signalfx/golib/event"
+	"github.com/signalfx/golib/sfxclient"
+	"github.com/sirupsen/logrus"
+	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/trace"
+)
+
+type SignalFXSink struct {
+	client   dpsink.Sink
+	hostname string
+	statsd   *statsd.Client
+	log      *logrus.Logger
+}
+
+// NewSignalFXSink creates a new Datadog sink for trace spans.
+func NewSignalFXSink(apiKey string, hostname string, stats *statsd.Client, log *logrus.Logger, client dpsink.Sink) (*SignalFXSink, error) {
+	if client == nil {
+		client := sfxclient.NewHTTPSink()
+		client.AuthToken = apiKey
+	}
+
+	return &SignalFXSink{
+		client:   client,
+		hostname: hostname,
+		statsd:   stats,
+		log:      log,
+	}, nil
+}
+
+// Name returns the name of this sink.
+func (sfx *SignalFXSink) Name() string {
+	return "signalfx"
+}
+
+// Start begins the sink. For SignalFx this is a noop.
+func (sfx *SignalFXSink) Start(traceClient *trace.Client) error {
+	return nil
+}
+
+// Flush sends metrics to SignalFx
+func (sfx *SignalFXSink) Flush(ctx context.Context, interMetrics []samplers.InterMetric) error {
+	span, _ := trace.StartSpanFromContext(ctx, "")
+	defer span.Finish()
```

Comment:
This ought to use the `sfr.trace` client structure from above and use `span.ClientFinish` so we can correctly use it in tests (: